codegen/src/test/resources/software/amazon/awssdk/codegen/poet/client/test-json-client-class.java

```diff
@@ -96,9 +98,14 @@ final class DefaultJsonClient implements JsonClient {
 
     private final SdkClientConfiguration clientConfiguration;
 
-    protected DefaultJsonClient(SdkClientConfiguration clientConfiguration) {
+    private final JsonServiceServiceClientConfiguration serviceClientConfiguration;
+
+    protected DefaultJsonClient(SdkClientConfiguration clientConfiguration,
+                                ClientOverrideConfiguration clientOverrideConfiguration) {
         this.clientHandler = new AwsSyncClientHandler(clientConfiguration);
         this.clientConfiguration = clientConfiguration;
+        this.serviceClientConfiguration = new JsonServiceServiceClientConfiguration(clientConfiguration,
+                                                                                    clientOverrideConfiguration);
```

Comment:
We should probably pass a copy of clientOverrideConfiguration, `clientOverrideConfiguration.toBuilder().build()`;

Reply:
+1

Reply:
Actually, since ClientOverrideConfiguration is already immutable, we probably don't need create a copy. I was thinking about v1, my bad!